crates/sla/src/lib.rs

```diff
@@ -421,16 +415,25 @@ impl<T: Config> Module<T> {
     /// * `amount` - the amount of polkabtc that was issued
     /// * `vault_id` - account of the vault
     fn _executed_issue_sla_change(amount: PolkaBTC<T>) -> Result<T::SignedFixedPoint, DispatchError> {
+        let amount_raw = Self::polkabtc_to_u128(amount)?;
+
         // update the number of issues performed
-        let mut count = <TotalIssueCount>::get();
-        count = count.checked_add(1).ok_or(Error::<T>::ArithmeticOverflow)?;
-        <TotalIssueCount>::set(count);
-
-        // TODO: fix this
-        let total = ext::treasury::get_total_supply::<T>();
-        let total_raw = Self::polkabtc_to_u128(total)?;
-        let average =
-            T::SignedFixedPoint::checked_from_rational(total_raw, count).ok_or(Error::<T>::TryIntoIntError)?;
+        let count = TotalIssueCount::mutate(|x| {
+            *x = x.saturating_add(1);
+            *x as u128
+        });
+        let total = LifetimeIssued::mutate(|x| {
+            *x = x.saturating_add(amount_raw);
+            *x
+        });
+
+        // calculate the average on raw input rather than in fixed_point - we don't want to fail
+        // if the result can not be losslessly represented. By using raw division we will be off
+        // but at most one Planck, which is acceptable. Note that the division can not fail, since
+        // we just incremented `count`, and being unsigned, it can not be 0 now.
+        let average_raw = total / count;
```

Comment:
We should probably use checked math even if it cannot fail.

Reply:
I think in this case it's really not necessary, but I'll change it so it can get merged